client_test.go

```diff
@@ -518,3 +582,24 @@ func TestCanNotDialInvalidAddress(t *testing.T) {
 		t.Fatal("invalid Connect() error")
 	}
 }
+
+func TestCanConnectToTwitch(t *testing.T) {
+	client := NewClient("justinfan123123", "oauth:123123132")
+
+	client.OnConnect(func() {
+		client.Disconnect()
+	})
+
+	client.Connect()
```

Comment:
We actually rely on Twitch here since this is a real connection to Twitch, but I think that's fine in a twitch irc library. Tests got like 1s slower though.